.github/workflows/api-checks.yml

```diff
@@ -0,0 +1,55 @@
+name: Template API Checks
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - app/**
+      - .github/workflows/api-checks.yml
+  pull_request:
+    paths:
+      - app/**
+      - .github/workflows/api-checks.yml
+
+defaults:
+  run:
+    working-directory: ./app
+
+env:
+  # Some commands in the Makefile check if this is set
+  # and add additional parameters to the commands called
+  CI: true
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: build container
+        run: make build
```

Comment:
Since we also split out `security` and `test` could we move `build` to a separate job? I think of checking that the build build works as different from linting.

Reply:
Not exactly - technically this could be removed. Every one of these actually runs the same docker build as they each recreate the docker container in order to actually run. If you look at the logs, you can see that they build and fetch packages first, which is why tests takes ~2minutes despite the actual tests running in about 10 seconds. there is a more complex way of building once and reusing across tasks not implemented here

Reply:
oh i see. in that case to keep things simple why don't we just run all the steps in one job so that they can all reuse the same build? and we can create a ticket in the future to [reuse build artifacts](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts).

Reply:
actually nvm, if running them in parallel is faster let's keep it that way